snuba/admin/audit_log/querylog.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import annotations
+
+from datetime import datetime
+from enum import Enum
+from functools import partial
+from typing import Any, Callable, Mapping, MutableMapping, Union
+
+import structlog
+
+from snuba.clickhouse.native import ClickhouseResult
+
+DATETIME_FORMAT = "%B %d, %Y %H:%M:%S %p"
+
+from snuba.admin.audit_log.base import AuditLog
+
+
+class QueryExecutionStatus(Enum):
+    SUCCEEDED = "succeeded"
+    FAILED = "failed"
+
+
+class QuerylogAuditLog(AuditLog):
+    def __init__(self) -> None:
+        self.logger = structlog.get_logger().bind(module=__name__)
+
+    def _record(
+        self, user: str, timestamp: str, action: str, data: Mapping[str, Any]
+    ) -> None:
+        query = data["query"]
+        status = data["status"]
```

Comment:
Would there be cases where some keys in `data` can be missing. You can probably make access to `data` keys safe at all the places you are using it.